File: ProjetAngular/src/app/Models/Cours.ts
import {Matieres} from "./Matieres";
import {Classes} from "./Classes";
import {AnneeScolaire} from "./AnneeScolaire";
import {Enseignants} from "./Enseignants";

export  class  Cours{

  constructor(
    public id: Number,
    public anneeScolaire: AnneeScolaire,
    public classe: Classes,
    public enseignant: Enseignants,
    public matiere: Matieres
  ) {}










 }

File: Projetemplois/src/main/java/com/emplois/temps/models/Cours.java
package com.emplois.temps.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Cours {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY )
    @Column(name="id")
    private Long id;

    @ManyToOne
    private  Matieres matieres;

    @ManyToOne
    private Classes classes;

    @ManyToOne
    private AnnéeScolaire annéeScolaire;

    @ManyToOne
    private Enseignants enseignants;

}
